integrations/go/workload-executor.go

```diff
@@ -0,0 +1,314 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"os/signal"
+	"reflect"
+	"syscall"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var emptyDoc = []byte{5, 0, 0, 0, 0}
+
+type driverWorkload struct {
+	Collection string
+	Database   string
+	TestData   []interface{} `bson:"testData"`
+	Operations []*operation
+}
+
+type operation struct {
+	Object    string
+	Name      string
+	Arguments bson.Raw
+	Result    interface{}
+}
+
+var specTestRegistry = bson.NewRegistryBuilder().
+	RegisterTypeMapEntry(bson.TypeEmbeddedDocument, reflect.TypeOf(bson.Raw{})).Build()
+
+func executeInsertOne(coll *mongo.Collection, args bson.Raw) (*mongo.InsertOneResult, error) {
+	doc := emptyDoc
+	opts := options.InsertOne()
+
+	elems, _ := args.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "document":
+			doc = val.Document()
+		default:
+			return nil, errors.New("unrecognized insertOne option: " + key)
```

Comment:
My understanding is that we'd end up recording this as an error so the `numErrors` field would be incremented and the test would fail. If this is the case, though, we lose the information about why it errored and we also risk having a false positive if the operation was actually supposed to error for some other reason. All of the `unrecognized...` default cases block us from even running the operation, so can we just change them to panic instead?